localstack/services/s3/provider.py

```diff
@@ -419,6 +669,285 @@ def is_object_expired(context: RequestContext, bucket: BucketName, key: ObjectKe
+    wildcard_pattern = f"{event_name[0:event_name.rindex(':')]}:*"
+    if wildcard_pattern in events:
+        return True
+    return False
+
+
+def _matching_filter(filter: Optional[NotificationConfigurationFilter], key_name: str) -> bool:
+    if not filter or not filter.get("Key", {}).get("FilterRules"):
+        return True
+    filter_rules = filter.get("Key").get("FilterRules")
+    for rule in filter_rules:
+        name = rule.get("Name", "").lower()
+        value = rule.get("Value", "")
+        if name == "prefix" and not key_name.startswith(value):
+            return False
+        if name == "suffix" and not key_name.endswith(value):
+            return False
+
+    return True
+
+
+def _send_event_to_event_bridge(
+    event_name: str,
+    bucket: moto_s3_models.FakeBucket,
+    key: moto_s3_models.FakeKey,
+    xray: str = None,
+):
+    s3api_client = aws_stack.connect_to_service("s3")
+    region = (
+        s3api_client.get_bucket_location(Bucket=bucket.name)["LocationConstraint"]
+        or config.DEFAULT_REGION
+    )
+    events_client = aws_stack.connect_to_service("events", region_name=region)
+    # structure defined here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/ev-events.html
+    entry = {
+        "Source": "aws.s3",
+        "Resources": [f"arn:aws:s3:::{bucket.name}"],
+        "Detail": {
+            "version": "0",
+            "bucket": {"name": bucket.name},
+            "object": {
+                "key": key.name,
+                "size": key.size,
+                "etag": key.etag.strip('"'),
+                "sequencer": "0062E99A88DC407460",
+            },
+            "request-id": "RKREYG1RN2X92YX6",
+            "requester": "074255357339",
+            "source-ip-address": "127.0.0.1",  # TODO previously headers.get("X-Forwarded-For", "127.0.0.1").split(",")[0]
+        },
+    }
+    # messages are bit different for EventBridge, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html
+    if "ObjectCreated" in event_name:
+        entry["DetailType"] = "Object Created"
+        event_type = event_name[event_name.rindex(":") + 1 :]
+        if event_type in ["Put", "Post", "Copy"]:
+            event_type = f"{event_type}Object"
+        entry["Detail"]["reason"] = event_type
+
+    if "ObjectRemoved" in event_name:
+        entry["DetailType"] = "Object Deleted"
+        entry["Detail"]["reason"] = "DeleteObject"
+        entry["Detail"]["deletion-type"] = "Permanently Deleted"
+        entry["Detail"]["object"].pop("etag")
+        entry["Detail"]["object"].pop("size")
+
+    if "ObjectTagging" in event_name:
+        entry["DetailType"] = "Object Tags Added" if "Put" in event_name else "Object Tags Deleted"
+
+    entry["Detail"] = json.dumps(entry["Detail"])
+
+    try:
+        events_client.put_events(Entries=[entry])
+    except Exception as e:
+        LOG.exception(f'Unable to send notification for S3 bucket "{bucket}" to EventBridge', e)
```

Comment:
no need to add the exception, `LOG.exception` uses the stacktrace to print the exception
```suggestion
        LOG.exception(f'Unable to send notification for S3 bucket "{bucket}" to EventBridge')
```